aggsender/db/types.go

```diff
@@ -37,21 +52,51 @@ type certificateInfo struct {
 	ExtraData               string                          `meddler:"extra_data"`
 }
 
-// toCertificate converts the certificateInfo struct to a Certificate struct
-func (c *certificateInfo) toCertificate() (*types.Certificate, error) {
-	signedCert := c.SignedCertificate
-
-	// If SignedCertificate contains a file path, read the content from the file
-	if signedCert != nil && *signedCert != "" && IsJSONFilePath(*signedCert) {
-		if content, err := os.ReadFile(*signedCert); err == nil {
-			contentStr := string(content)
-			signedCert = &contentStr
-		} else {
-			log.Errorf("Failed to read signed certificate file %s: %v", *signedCert, err)
-			return nil, err
+// SetSignedCertificateFilename sets the SignedCertificate field to external file
+func (c *certificateInfo) SetSignedCertificateFilename(filename string) {
+	if c == nil {
+		return
+	}
+	c.SignedCertificate = nil
+	if filename != "" {
+		c.SignedCertificate = new(string)
+		*c.SignedCertificate = PrefixFilename + filename
+	}
+}
+
+// SignedCertificateFilename returns the filename if SignedCertificate is a file reference
+func (c *certificateInfo) SignedCertificateFilename() *string {
+	if c == nil || c.SignedCertificate == nil {
+		return nil
+	}
+	if strings.HasPrefix(*c.SignedCertificate, PrefixFilename) {
+		filename := strings.TrimPrefix(*c.SignedCertificate, PrefixFilename)
+		return &filename
+	}
```

Comment:
The suggestion came from the IDE, I'd say pretty neat replacement. WDYT?
```suggestion
	if after, ok := strings.CutPrefix(*c.SignedCertificate, PrefixFilename); ok {
		filename := after
		return &filename
	}
```

Reply:
For me is more clear the current version